lib/ui/widgets/track/track_list_item.dart

```diff
@@ -20,97 +30,166 @@ class TrackListItem extends StatelessWidget {
+      confirmDismiss: (direction) =>
+          _confirmDismiss(context, localizations, theme),
+      onDismissed: (direction) => onDismissed(),
+            child: ClickableTile(
+              onTap: () => Navigator.push(
+                context,
+                MaterialPageRoute(
+                  builder: (context) => TrackDetailScreen(track: track),
+                ),
+              ),
+          child: hasGeolocations
+              ? _buildTrackContent(context, localizations, theme, track)
+              : _buildNoGeolocationsContent(context, localizations, theme),
+            )));
+  }
+
+  Widget _buildDismissBackground(ThemeData theme) {
+    return Container(
+      color: theme.colorScheme.error,
+      alignment: Alignment.centerRight,
+      padding: const EdgeInsets.only(right: spacing),
+      child: const Icon(Icons.delete_outline),
+    );
+  }
+
+  Future<bool?> _confirmDismiss(
+      BuildContext context, AppLocalizations localizations, ThemeData theme) {
+    return showDialog<bool>(
+      context: context,
+      builder: (BuildContext context) {
+        return AlertDialog(
+          title: Text(localizations.trackDelete),
+          content: Text(localizations.trackDeleteConfirmation),
+          actions: <Widget>[
+            TextButton(
+              onPressed: () => Navigator.of(context).pop(false),
+              child: Text(localizations.generalCancel),
+            ),
+            FilledButton(
+              style: ButtonStyle(
+                backgroundColor: WidgetStateProperty.all(
+                  theme.colorScheme.error,
+                ),
+              ),
+              onPressed: () => Navigator.of(context).pop(true),
+              child: Text(localizations.generalDelete),
+            ),
+          ],
+        );
+      },
+    );
+  }
+
+  Widget _buildNoGeolocationsContent(
+      BuildContext context, AppLocalizations localizations, ThemeData theme) {
+    return Center(
+      child: Text(
+        localizations.trackNoGeolocations,
+        style: theme.textTheme.bodyMedium
+            ?.copyWith(color: theme.colorScheme.error),
+      ),
+    );
+  }
+
+  Widget _buildTrackContent(
+      BuildContext context,
+      AppLocalizations localizations, ThemeData theme, TrackData track) {
+    final date =
+        DateFormat('dd.MM.yyyy').format(track.geolocations.first.timestamp);
+    final trackStart =
+        DateFormat('HH:mm').format(track.geolocations.first.timestamp);
+    final trackEnd =
+        DateFormat('HH:mm').format(track.geolocations.last.timestamp);
+    final times = '$trackStart - $trackEnd';
+    final duration = localizations.generalTrackDurationShort(
+        track.duration.inHours.toString(),
+        track.duration.inMinutes
+            .remainder(60)
+            .toString()
+            .padLeft(2, '0')); // Ensure 2-digit format
+    final distance = localizations
+        .generalTrackDistance((track.distance / 1000).toStringAsFixed(2));
```

Comment:
`track.distance` already returns kilometers. no need to divide when showing it as km